src/nip.py

```diff
@@ -131,28 +131,18 @@ def handle_self(self, name):
 
     def handle_subdomains(self, qname):
         subdomain = qname[0:qname.find(self.domain) - 1]
-        # Split foo.bar.10-0-0-1 in parts
-        subdomain_parts = subdomain.split('.')
+        match = re.findall('^(?:.+\.)?(\d{1,3}[-.]\d{1,3}[-.]\d{1,3}[-.]\d{1,3})$', subdomain)
```

Comment:
I would suggest supporting either `-` or `.` but not both at the same time. This could be done using back-references: `^(?:.+\.)?(\d{1,3}([-.])\d{1,3}\2\d{1,3}\2\d{1,3})$` https://regex101.com/r/Tamzd5/2

Reply:
I would love to, but I will be very honest in saying that I have no idea how to and neither do I have any way of testing the change before creating a PR.

Reply:
@Doqnach
Thank you, your suggestion is very reasonable. So I picked it in f0640ad.